tests/tests_suite_test.go

```diff
@@ -134,3 +152,46 @@ func BuildTestSuite() {
 	})
 
 }
+
+func updateAAQNamespaceSelector(aaqClient *clientset.Clientset, k8sClient *kubernetes.Clientset, selector *metav1.LabelSelector) (originalSelector *metav1.LabelSelector, err error) {
+	// Ensure AAQ is deployed and overwrite default namespace selector to target any namespace
+	aaqList, err := aaqClient.AaqV1alpha1().AAQs().List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("cannot list AAQ objects")
+	}
+
+	if len(aaqList.Items) != 1 {
+		return nil, fmt.Errorf("expecting a single AAQ instance, actual: %d", len(aaqList.Items))
+	}
+
+	aaqWebhook, err := k8sClient.AdmissionregistrationV1().MutatingWebhookConfigurations().Get(context.Background(), cluster.MutatingWebhookConfigurationName, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	originalAAQ := aaqList.Items[0]
+	originalWebhookSelector := aaqWebhook.Webhooks[0].NamespaceSelector
+
+	Eventually(func() error {
+		aaq := originalAAQ.DeepCopy()
+		aaq.Spec.NamespaceSelector = selector
+
+		_, err = aaqClient.AaqV1alpha1().AAQs().Update(context.Background(), aaq, metav1.UpdateOptions{})
+		return err
+	}).WithTimeout(30*time.Second).WithPolling(time.Second).ShouldNot(HaveOccurred(), "cannot update AAQ object's namespace selector'")
+
+	Eventually(func() error {
+		aaqWebhook, err := k8sClient.AdmissionregistrationV1().MutatingWebhookConfigurations().Get(context.Background(), cluster.MutatingWebhookConfigurationName, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+
+		if reflect.DeepEqual(aaqWebhook.Webhooks[0].NamespaceSelector, originalWebhookSelector) {
+			return fmt.Errorf("expecting namespace selector to be updated")
+		}
+
+		return nil
+	}).WithTimeout(30*time.Second).WithPolling(time.Second).ShouldNot(HaveOccurred(), fmt.Sprintf("webhook namespace selector is not updated"))
+
+	return originalAAQ.Spec.NamespaceSelector, nil
+}
```

Comment:
Can you please save the original AAQ outside of this function? I find it a bit confusing that this function returns the original AAQ.

Reply:
Done!